### PR TITLE
Add initial Spanish language support and move German, Norwegian and Italian to CP1252 generation function

### PR DESCRIPTION
**Pull request by zenseii**
This  adds a Spanish language option to the game and font support.

The existing Spanish translation is 54 % complete.

Spanish uses CP1252 so I added its letters in a new function `generateCP1252Alphabet()`.

Tidied up so that German, Norwegian and Italian all use CP1252 too. French is supported by CP1252 as well, but the OG encoding does not support that. A decision on how to deal with this is necessary.

Some letter generation improvements to existing CP1252 letters like lowercase `ì`.

All Spanish characters have been implemented.

Current progress:
image
image
image

image

### CODE REVIEW

**Comment by zenseii** [comment]
Hi @ihhub ! This PR is ready for you review when you should have the time.

All characters necessary for Spanish have been added.

I suggest to move the French characters in a separate PR as it will require some more attention to preserve OG encoding compatibility.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @zenseii , I left one optimization suggestion here. Would you mind check it in your free time?

File: src/fheroes2/gui/ui_font.cpp [hunk 1]
```diff
@@ -1452,6 +1283,41 @@ namespace
         {
             std::vector<fheroes2::Sprite> & font = icnVsSprite[ICN::FONT];
 
+            // Inverted exclamation mark !.
+            font[161 - 32].resize( font[33 - 32].width() + 1, font[33 - 32].height() + 3 );
+            font[161 - 32].reset();
+            fheroes2::Copy( font[33 - 32], 1, 0, font[161 - 32], 1, 3, font[33 - 32].width(), font[33 - 32].height() );
+            {
+                // TODO: add proper Flip function variant.
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:1291]: This comment is not needed since it works fine with full images.

File: src/fheroes2/gui/ui_font.cpp [hunk 2]
```diff
@@ -1452,6 +1283,41 @@ namespace
...
+            }
+            font[161 - 32].setPosition( font[33 - 32].x(), font[33 - 32].y() + 2 );
+            updateNormalFontLetterShadow( font[161 - 32] );
+
+            // Inverted question mark ?. Replaced with normal for now.
+            font[191 - 32].resize( font[63 - 32].width() + 1, font[63 - 32].height() );
+            font[191 - 32].reset();
+            fheroes2::Copy( font[63 - 32], 1, 0, font[191 - 32], 0, 0, font[63 - 32].width(), 11 );
+            {
+                // TODO: add proper Flip function variant.
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:1303]: This comment is not needed since it works fine with full images.

File: src/fheroes2/gui/ui_font.cpp [hunk 3]
```diff
@@ -1534,30 +1633,68 @@ namespace
...
+            }
+            font[161 - 32].setPosition( font[33 - 32].x(), font[33 - 32].y() + 2 );
+            updateSmallFontLetterShadow( font[161 - 32] );
+
+            // Inverted question mark ?. Replaced with normal for now.
+            font[191 - 32].resize( font[63 - 32].width(), font[63 - 32].height() );
+            font[191 - 32].reset();
+            fheroes2::Copy( font[63 - 32], 1, 0, font[191 - 32], 0, 0, font[63 - 32].width(), 7 );
+            {
+                // TODO: add proper Flip function variant.
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:1687]: This comment is not needed since it works fine with full images.

File: src/fheroes2/gui/ui_font.cpp [hunk 4]
```diff
@@ -1534,30 +1633,68 @@ namespace
...
         // Small font.
         {
             std::vector<fheroes2::Sprite> & font = icnVsSprite[ICN::SMALFONT];
 
+            // Inverted exclamation mark !.
+            font[161 - 32].resize( font[33 - 32].width(), font[33 - 32].height() );
+            font[161 - 32].reset();
+            fheroes2::Copy( font[33 - 32], 1, 0, font[161 - 32], 1, 0, font[33 - 32].width(), 7 );
+            {
+                // TODO: add proper Flip function variant.
```
Comment (ihhub) [inline src/fheroes2/gui/ui_font.cpp:1675]: This comment is not needed since it works fine with full images.

**Review by ihhub — COMMENTED** [review]
Hi @zenseii ,  I left very tiny suggestions about comments. Could you please check again?

**Review by ihhub — APPROVED** [review]

**Comment by ihhub** [comment]
@zenseii , thank you so much for these font related changes!

**Pull request merged**